test/unit/test_array.cxx

```diff
@@ -436,6 +436,29 @@ void test_array_generate_empty_strings()
     "Array of 12 empty strings came out wrong.");
 }
 
+void test_sparse_arrays()
+{
+  // Reproduce #922 : NULL not paying for its separator in an array, causing
+  // problems in sparse arrays
+
+  PQXX_CHECK_EQUAL(
+    pqxx::to_string(std::vector<std::optional<int>>{
+      std::nullopt, std::nullopt, std::nullopt, std::nullopt}),
+    "{NULL,NULL,NULL,NULL}",
+    "Array of optional<int> filled with std::nullopt came out wrong");
```

Comment:
Perhaps we can improve this further by putting the expected output in a variable.  Then we can do separate checks for...
1. The conversion produces the expected output (as here).
2. The _budget_ for the conversion is less than the output's size, or equal to it.

(In fact if that's enough to detect the bug, perhaps we won't even need the check with the larger array.)

Reply:
Fair point!